test/s3.js

```diff
@@ -5,7 +5,12 @@ var s3 = require('../src/js/s3');
 var fs = require('fs');
 var debug = require('gulp-debug');
 var through2 = require('through2').obj;
-var AWS = require('../node_modules/gulp-awspublish/node_modules/aws-sdk');
+var awspath = '../node_modules/gulp-awspublish/node_modules/aws-sdk';
+if (fs.existsSync(__dirname+'/../node_modules/aws-sdk/package.json')) {
+  awspath = '../node_modules/aws-sdk';
+}
+var AWS = require(awspath);
```

Comment:
It would require `lucify-component-builder` to depend on `aws-sdk` directly. Which it maybe should do anyway.

Then you can do like this:
```
var awspath = require.resolve('aws-sdk').replace('/lib/aws.js', '')
```

This however breaks if `aws-sdk` changes its entry point from '/lib/aws.js' to something else. One could probably do a more clever trick to strip out that part of the path.